indexers/keyvalue/MongoDBIndexer/__init__.py

```diff
@@ -46,13 +44,15 @@ def get_create_handler(self):
     def get_query_handler(self):
         return self.handler
     
-    def add(self, keys: Iterator[int], values: Iterator[bytes], *args, **kwargs):
-        mongo_docs = [{'_id': i, 'values': j} for i, j in zip(keys, values)]
-        
+    def add(self, keys: Iterator[int], values: Iterator[bytes], *args, **kwargs) -> None:
+        total_inserted_ids = []
         with self.write_handler as mongo_handler:
-            inserted_ids = mongo_handler.insert(documents=mongo_docs)
+            for i, j in zip(keys, values):
+                doc = {'_id': i, 'values': j}
+                inserted_ids = mongo_handler.insert(documents=[doc])
```

Comment:
why aren't we inserting in batch? it seems mongo_handler accepts more than one document to be inserted at once right?

Reply:
https://github.com/jina-ai/jina-hub/pull/1232#discussion_r537803852

Reply:
Smaller memory footprint

Reply:
at the cost of possibly more network IO

Reply:
I would definitely go for `batching`

Reply:
there is some `batching_multi_input` that could serve this purpose